app/earn/[...vault]/hooks/use-vault-deposit-change.ts

```diff
@@ -0,0 +1,32 @@
+import { useMemo } from 'react'
+import { useVaultContext } from './use-vault-context'
+import { useAppSelector } from '@/store/hooks'
+import { FormAction } from '@/store/slices/vaultActionSlice'
+import { getAmountInUSD } from '@/shared'
+import type { Vault } from '@/api'
+
+export function useVaultDeposit() {
+  const { vaultBalanceBN } = useAppSelector(state => state.vaultUser)
+  return useDeposit(vaultBalanceBN)
+}
+
+export function useVaultDepositUSD(vault: Vault) {
+  const [total, change] = useVaultDeposit()
+  const [depositInUSD, priceDecimals] = getAmountInUSD(total, vault)
+  const changeInUSD = getAmountInUSD(change, vault)
+  return { depositInUSD, changeInUSD, deposit: total, decimals: priceDecimals }
+}
+
+function useDeposit(amount: bigint | string) {
```

Comment:
Probably we need start to write tests for this logic. It already pretty complicated and I scary to change something and break everything